detection_rules/kbwrap.py

```diff
@@ -71,6 +74,96 @@ def upload_rule(ctx, rules, replace_id):
     return results
 
 
+@kibana_group.command('import-rules')
+@multi_collection
+@click.option('--overwrite', '-o', is_flag=True, help='Overwrite existing rules')
+@click.option('--overwrite-exceptions', '-e', is_flag=True, help='Overwrite exceptions in existing rules')
+@click.option('--overwrite-action-connectors', '-a', is_flag=True,
+              help='Overwrite action connectors in existing rules')
+@click.pass_context
+def kibana_import_rules(ctx: click.Context, rules: RuleCollection, overwrite: Optional[bool] = False,
+                        overwrite_exceptions: Optional[bool] = False,
+                        overwrite_action_connectors: Optional[bool] = False) -> (dict, List[RuleResource]):
+    """Import custom rules into Kibana."""
+    kibana = ctx.obj['kibana']
+    rule_dicts = [r.contents.to_api_format() for r in rules]
+    with kibana:
+        response, successful_rule_ids, results = RuleResource.import_rules(
+            rule_dicts,
+            overwrite=overwrite,
+            overwrite_exceptions=overwrite_exceptions,
+            overwrite_action_connectors=overwrite_action_connectors
+        )
+
+    if successful_rule_ids:
+        click.echo(f'{len(successful_rule_ids)} rule(s) successfully imported')
+        rule_str = '\n - '.join(successful_rule_ids)
+        print(f' - {rule_str}')
+    if response['errors']:
+        click.echo(f'{len(response["errors"])} rule(s) failed to import!')
+        for error in response['errors']:
+            click.echo(f' - {error["rule_id"]}: ({error["error"]["status_code"]}) {error["error"]["message"]}')
+
+    return response, results
+
+
+@kibana_group.command('export-rules')
+@click.option('--directory', '-d', required=True, type=Path, help='Directory to export rules to')
+@click.option('--rule-id', '-r', multiple=True, help='Optional Rule IDs to restrict export to')
+@click.option('--skip-errors', '-s', is_flag=True, help='Skip errors when exporting rules')
+@click.pass_context
+def kibana_export_rules(ctx: click.Context, directory: Path,
+                        rule_id: Optional[Iterable[str]] = None, skip_errors: bool = False) -> List[TOMLRule]:
+    """Export custom rules from Kibana."""
+    kibana = ctx.obj['kibana']
+    with kibana:
+        results = RuleResource.export_rules(list(rule_id))
+
+    if results:
+        directory.mkdir(parents=True, exist_ok=True)
+
+    errors = []
+    exported = []
+    for rule_resource in results:
+        try:
+            contents = TOMLRuleContents.from_rule_resource(rule_resource, maturity='production')
+            threat = contents.data.get('threat')
+            first_tactic = threat[0].tactic.name if threat else ''
+            rule_name = rulename_to_filename(contents.data.name, tactic_name=first_tactic)
+            rule = TOMLRule(contents=contents, path=directory / f'{rule_name}')
```

Comment:
So I think I figured out the issue with some of the validation errors and its in this new method you created.

Since you dont go through the `rule_prompt`, you miss out on some of the transforms we had to do to address incorrectly written schemas like `new_terms`. 

We recently fixed this by adding a transformation here https://github.com/elastic/detection-rules/blob/main/detection_rules/cli_utils.py#L168-L178 which is used by other similar methods that create toml files `create_rule` and `import_rules_into_repo`

Reply:
The core problem there is that we store it in incorrect schema form. The real fix here is to update the dataclass to reflect the static schema. This is out of scope of this PR

Reply:
related to #3296